File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.swift
var input = readLine()!.split(separator: " ").compactMap{Int($0)}
var n = input[0]
var k = input[1]

var queue = [Int]()
queue.append(n)

var idx = 0
var visited = Array(repeating: 0, count: 100001)
visited[n] = 1
var solved = false
if (n==k){
    print(0)
}else{
    while(idx<queue.count){
        if(solved) {break}
        var now = queue[idx]
        var nextArr = [now-1,now+1,now*2]
        idx+=1
        for next in nextArr{
            if(0...100000 ~= next && visited[next]==0){
                    queue.append(next)
                    visited[next] = visited[now] + 1
                if(next == k){
                    print(visited[next]-1)
                    solved = true
                    break
                }
                }
        }
    }

}

File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.py
from collections import deque

n, k = map(int, input().split())
graph = [-1] * 100001
dq = [1, -1, 2]

def bfs(start):
    queue = deque()
    queue.append(start)
    graph[start] = 0
    
    while queue:
        q = queue.popleft()
        
        for i in dq:
            nq = q * 2 if i == 2 else q + i
            if 0 <= nq <= 100000 and graph[nq] == -1:
                queue.append(nq)
                graph[nq] = graph[q] + 1
            if nq == k:
                return graph[nq]

print(bfs(n))